### PR TITLE
Add synchrony metrics to quality metrics

### PR DESCRIPTION
**Pull request by Moritz-Alexander-Kern**
This PR is adding a metric for the presence of synchronous spiking events across multiple spike trains.

This is used to characterize synchronous events within the same spike train and across different spike
trains. This way synchronous events can be found both in multi-unit and single-unit spike trains.

The code was adapted from  [Elephant - Electrophysiology Analysis Toolkit](https://github.com/NeuralEnsemble/elephant/blob/master/elephant/spike_train_synchrony.py), see docs for more information 
[Spike train synchrony](https://elephant.readthedocs.io/en/latest/reference/_spike_train_processing.html#module-elephant.spike_train_synchrony)

### CODE REVIEW

**Comment by alejoe91** [comment]
Thanks @Moritz-Alexander-Kern! I'll provide some feedback soon

File: spikeinterface/qualitymetrics/misc_metrics.py [hunk 1]
```diff
@@ -517,3 +521,85 @@ def _compute_rp_violations_numba(nb_rp_violations, spike_trains, spike_clusters,
...
+    # free some memory
+    synchrotool.complexity_histogram = []
+    synchrotool.time_histogram = []
+    # annotate synchrofacts
+    synchrotool.annotate_synchrofacts()
+
+    # create a dictionary 'synchrony_metrics'
+    synchrony_metrics = {
+        # create a dictionary for each synchrony_size
+        f'syncSpike_{synchrony_size}': {
```
Comment (alejoe91) [inline spikeinterface/qualitymetrics/misc_metrics.py:562]: Use snake case

**Review by alejoe91 — CHANGES_REQUESTED** [review]
First round of feedback! Looks good, but I'd rather have the low-level implementation of the `annotate_synchrofacts` here rather than depending on elephant just for that function.

Could you also add tests in the `quality_metrics/tests/test_metric_functions.py`?

File: spikeinterface/qualitymetrics/misc_metrics.py (hunk 1 above)
Comment (Moritz-Alexander-Kern) [reply spikeinterface/qualitymetrics/misc_metrics.py:562]:
```suggestion
        f'sync_spike_{synchrony_size}': {
```

File: pyproject.toml [hunk 2]
```diff
@@ -26,6 +26,7 @@ dependencies = [
     "threadpoolctl",
     "tqdm",
     "probeinterface>=0.2.14",
+    "elephant>=0.11.0"
```
[the remark this answers is not in the document]
Comment (Moritz-Alexander-Kern) [reply pyproject.toml:29]: Thanks for the feedback, I see your point.

Please allow me some time to look into this.

[the remark this answers is not in the document]
Comment (alejoe91) [reply pyproject.toml:29]: Of course :) no rush!

**Comment by alejoe91** [comment]
@Moritz-Alexander-Kern @musall any update on this?

**Comment by Moritz-Alexander-Kern** [comment]
Update: Currently porting the code from elephant. 

Following your suggestion to port the code from elephant to spikeinterface. I'm in the process of implementing this change and will update you once it is complete.

**Comment by alejoe91** [comment]
Hi @Moritz-Alexander-Kern 

We recently changed the main branch to `main` instead of `master`.

We should not port the full `elephant` classes here, but just the minimal function required to find synchronous events (`annotate_synchrofacts`), which should take numpy array as inputs. 

Would you have time to work on this?

**Comment by alejoe91** [comment]
Closing in favor of #1951

**Pull request closed without merging**